schema/reportdb/reportdb-user-management.sh

```diff
@@ -0,0 +1,177 @@
+#! /bin/bash
+
+DBNAME="reportdb"
+INTERACTIVE=1
+INTERACTIVE_RETRIES=3
+
+if [ 0$UID -gt 0 ]; then
+    echo Run as root.
+    exit 1
+fi
+
+
+print_help() {
+    cat <<HELP
+usage: reportdb-user-management.sh [options]
+
+options:
+  --help 
+            show this help message and exit
+  --non-interactive
+            Switches to non-interactive mode
+  --dbuser=DBUSER
+            Report DB User
+  --dbpassword=DBPASSWORD
+            Report DB Password
+  --add
+            Add the new user
+  --delete
+            Delete the user
+  --modify
+            Set a new password
+
+HELP
+    exit 1
+}
+
+default_or_input() {
+    local MSG="$1"
+    local VARIABLE="$2"
+    local DEFAULT="$3"
+    local SILENT="$4"
+
+    local INPUT
+    local CURRENT_VALUE=${!VARIABLE}
+    #in following code is used not so common expansion
+    #var_a=${var_b:-word}
+    #which is like: var_a = $var_b ? word
+    DEFAULT=${CURRENT_VALUE:-$DEFAULT}
+    local VARIABLE_ISSET=$(set | grep "^$VARIABLE=")
+
+    echo -n "$MSG [$DEFAULT]: "
+    if [ "$INTERACTIVE" = "1" -a  -z "$VARIABLE_ISSET" ]; then
+        MANUAL_ANSWERS=1
+        if [ "$SILENT" = 1 ]; then
+          read -s INPUT
+        else
+          read INPUT
+        fi
+    elif [ -z "$VARIABLE_ISSET" ]; then
+        echo "$DEFAULT"
+    else
+        DEFAULT=${!VARIABLE}
+        echo "$DEFAULT"
+    fi
+    if [ -z "$INPUT" ]; then
+        if [ "$DEFAULT" = "y/N" -o "$DEFAULT" = "Y/n" ]; then
```

Comment:
From what I knew `-a` and `-o` are kind of deprecated. `test` man page says: 

> Binary -a and -o are inherently ambiguous.  Use 'test EXPR1 && test EXPR2' or 'test EXPR1 || test EXPR2' instead.

I don't think it's strictly needed in this use case but I would suggest to use `[ "$DEFAULT" = "y/N" ] || [ "$DEFAULT" = "Y/n" ]`. Same for the `-a` above.